File: src/FileTransferHandler.java
import java.awt.datatransfer.DataFlavor;
import java.awt.datatransfer.UnsupportedFlavorException;
import java.io.File;
import java.io.IOException;
import java.util.List;
import javax.swing.TransferHandler;

public class FileTransferHandler extends TransferHandler {

    private final ClassySharkBytecodeViewer classySearch;

    public FileTransferHandler(ClassySharkBytecodeViewer classySearch) {
        this.classySearch = classySearch;
    }

    public boolean canImport(TransferSupport ts) {
        return true;
    }

    public boolean importData(TransferSupport ts) {
        try {
            @SuppressWarnings("rawtypes")
            List data = (List) ts.getTransferable().getTransferData(
                    DataFlavor.javaFileListFlavor);
            if (data.size() < 1) {
                return false;
            }

            for (Object item : data) {
                File file = (File) item;
                classySearch.fileDragged(file);
            }

            return true;

        } catch (UnsupportedFlavorException e) {
            return false;
        } catch (IOException e) {
            return false;

        }
    }
}

File: src/FileTransferHandler.kt
import java.awt.datatransfer.DataFlavor
import java.awt.datatransfer.UnsupportedFlavorException
import java.io.File
import java.io.IOException
import javax.swing.TransferHandler

class FileTransferHandler(private val classySharkBytecodeViewer: ClassySharkBytecodeViewer) : TransferHandler() {

    override fun canImport(ts: TransferHandler.TransferSupport): Boolean {
        return true
    }

    override fun importData(ts: TransferHandler.TransferSupport): Boolean {
        try {
            val data = ts.transferable.getTransferData(
                    DataFlavor.javaFileListFlavor) as List<*>
            if (data.size < 1) {
                return false
            }

            data.map { it as File }.forEach { classySharkBytecodeViewer.onFileDragged(it) }

            return true

        } catch (e: UnsupportedFlavorException) {
            return false
        } catch (e: IOException) {
            return false

        }
    }
}